Card/src/main/java/com/paypal/android/card/api/CardAPI.kt

```diff
@@ -1,23 +1,52 @@
-package com.paypal.android.card
+package com.paypal.android.card.api
 
+import com.paypal.android.card.Card
+import com.paypal.android.card.ConfirmPaymentSourceRequestFactory
+import com.paypal.android.card.CreateOrderRequestFactory
+import com.paypal.android.card.OrderRequest
+import com.paypal.android.card.threedsecure.ThreeDSecureRequest
 import com.paypal.android.core.API
+import com.paypal.android.core.API.Companion.SUCCESSFUL_STATUS_CODES
 import com.paypal.android.core.APIClientError
+import com.paypal.android.core.APIRequest
 import com.paypal.android.core.HttpResponse.Companion.SERVER_ERROR
 import com.paypal.android.core.HttpResponse.Companion.STATUS_UNDETERMINED
 import com.paypal.android.core.HttpResponse.Companion.STATUS_UNKNOWN_HOST
 import com.paypal.android.core.OrderErrorDetail
-import com.paypal.android.core.OrderStatus
 import com.paypal.android.core.PayPalSDKError
 import com.paypal.android.core.PaymentsJSON
-import java.net.HttpURLConnection.HTTP_OK
 
 internal class CardAPI(
     private val api: API,
-    private val requestFactory: CardAPIRequestFactory = CardAPIRequestFactory()
 ) {
+
     @Throws(PayPalSDKError::class)
-    suspend fun confirmPaymentSource(orderID: String, card: Card): CardResult {
-        val apiRequest = requestFactory.createConfirmPaymentSourceRequest(orderID, card)
+    suspend fun confirmPaymentSource(
+        orderID: String,
+        card: Card,
+        threeDSecureRequest: ThreeDSecureRequest? = null
+    ): ConfirmPaymentSourceResponse {
+        return performRequest(
+            ConfirmPaymentSourceRequestFactory.createRequest(orderID, card, threeDSecureRequest),
+            ConfirmPaymentSourceRequestFactory::parseResponse
+        )
+    }
+
+    @Throws(PayPalSDKError::class)
+    suspend fun createOrder(
+        orderRequest: OrderRequest,
+        threeDSecureRequest: ThreeDSecureRequest? = null
+    ): CreateOrderResponse {
+        return performRequest(
+            CreateOrderRequestFactory.createRequest(orderRequest, threeDSecureRequest),
+            CreateOrderRequestFactory::parseResponse
+        )
+    }
+
+    private suspend fun <R> performRequest(
+        apiRequest: APIRequest,
+        parseResult: (body: String, correlationId: String?) -> R
+    ): R {
```

Comment:
Do we need generics here?

Reply:
pefroming a request is kinda mechanical, so the idea is that you supply an `APIRequest` to the method, and them a function that parses the result and return the correct `response`. In `confirm_payment_source` it returns a `ConfirmPaymentSourceResponse` and in create order a `CreateOrderResponse`. I think it will be helpfull when we add more api requests to the module